zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/client/command/UserTaskListenersTest.java

```diff
@@ -223,6 +228,85 @@ void shouldRejectCompleteTaskListenerJobCompletionWhenVariablesAreSet() {
         });
   }
 
+  @Test
+  public void shouldRejectUserTaskCompletionWhenTaskListenerDeniesTheWork() {
+    // given
+    final var listenerType = "my_listener";
+    final var userTaskKey =
+        resourcesHelper.createSingleUserTask(
+            t -> t.zeebeTaskListener(l -> l.complete().type(listenerType)));
+
+    final JobHandler completeJobHandler =
+        (jobClient, job) -> client.newCompleteCommand(job).result().denied(true).send().join();
+    final var recordingHandler = new RecordingJobHandler(completeJobHandler);
+
+    client.newWorker().jobType(listenerType).handler(recordingHandler).open();
+
+    // when: invoke complete user task command
+    final ZeebeFuture<CompleteUserTaskResponse> completeUserTaskFuture =
+        client.newUserTaskCompleteCommand(userTaskKey).send();
+
+    // TL job should be successfully completed with the result "denied" set correctly
+    ZeebeAssertHelper.assertJobCompleted(
+        listenerType,
+        (userTaskListener) -> {
+          assertThat(userTaskListener.getResult().isDenied()).isTrue();
+        });
+
+    final var rejectionReason =
+        String.format(
+            "Command 'COMPLETE' rejected with code 'INVALID_STATE': Completion of the User Task with key '%s' was rejected by Task Listener",
+            userTaskKey);
+
+    //     verify the rejection
+    final ProblemException problemException =
+        catchThrowableOfType(ProblemException.class, completeUserTaskFuture::join);
+    assertThat(problemException.details().getTitle()).isEqualTo(RejectionType.INVALID_STATE.name());
+    assertThat(problemException.details().getDetail()).isEqualTo(rejectionReason);
+
+    // verify the expected sequence of User Task intents
+    assertUserTaskIntentsSequence(
+        UserTaskIntent.COMPLETING,
+        UserTaskIntent.DENY_TASK_LISTENER,
+        UserTaskIntent.COMPLETION_DENIED);
+  }
+
+  @Test
+  public void shouldCompleteUserTaskWhenTaskListenerAcceptsTheWork() {
```

Comment:
I have the same doubts. @korthout what would be your thoughts on that?
On one side we'd want to test all scenarios, but on the other side it might be enough to test setting the result explicitly.

Reply:
I'll make the change, thank you 👍